magicblock-task-scheduler/src/service.rs

```diff
@@ -238,4 +248,71 @@ impl TaskSchedulerService {
 
         Ok(())
     }
+
+    async fn run(
+        self,
+        mut context_sub: Receiver<GeyserMessage>,
+        token: CancellationToken,
+    ) {
+        let mut interval = tokio::time::interval(self.tick_interval);
+        let db = self.db.clone();
```

Comment:
nit: why clone the fields here? And by extension is it possible to use associated method which accept `self`, instead of using associated functions, which require passing the fields explicitly. It's just less verbose and cleaner IMHO